src/integration/helpers_test.go

```diff
@@ -0,0 +1,146 @@
+package integration_test
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"code.cloudfoundry.org/tlsconfig/certtest"
+	colmetricspb "go.opentelemetry.io/proto/otlp/collector/metrics/v1"
+	commonpb "go.opentelemetry.io/proto/otlp/common/v1"
+	metricspb "go.opentelemetry.io/proto/otlp/metrics/v1"
+	"google.golang.org/grpc/credentials"
+)
+
+func TempFilePath(name string) string {
+	dir := fmt.Sprintf("./tmp-%d", GinkgoParallelProcess())
+	os.MkdirAll(dir, 0777)
+	DeferCleanup(os.RemoveAll, dir)
+	return filepath.Join(dir, name)
+}
+
+func NewSimpleResourceMetrics(name string) metricspb.ResourceMetrics {
+	return metricspb.ResourceMetrics{
+		ScopeMetrics: []*metricspb.ScopeMetrics{
+			{
+				Metrics: []*metricspb.Metric{
+					{
+						Name: name,
+						Data: &metricspb.Metric_Sum{
+							Sum: &metricspb.Sum{
+								AggregationTemporality: metricspb.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE,
+								IsMonotonic:            true,
+								DataPoints: []*metricspb.NumberDataPoint{
+									{
+										TimeUnixNano: uint64(0),
+										Attributes: []*commonpb.KeyValue{
+											{
+												Key:   "instance_id",
+												Value: &commonpb.AnyValue{Value: &commonpb.AnyValue_StringValue{StringValue: "dead-beef"}},
+											},
+										},
+										Value: &metricspb.NumberDataPoint_AsInt{
+											AsInt: int64(42),
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+type FakeMetricsServiceServer struct {
+	colmetricspb.MetricsServiceServer
+	ExportMetricsServiceRequests chan *colmetricspb.ExportMetricsServiceRequest
+}
+
+func NewFakeMetricsServiceServer() FakeMetricsServiceServer {
+	return FakeMetricsServiceServer{
+		ExportMetricsServiceRequests: make(chan *colmetricspb.ExportMetricsServiceRequest, 10),
+	}
+}
+
+func (s FakeMetricsServiceServer) Export(ctx context.Context, emsr *colmetricspb.ExportMetricsServiceRequest) (*colmetricspb.ExportMetricsServiceResponse, error) {
+	s.ExportMetricsServiceRequests <- emsr
+	return &colmetricspb.ExportMetricsServiceResponse{}, nil
+}
+
+type OTELConfigVars struct {
+	IngressOTLPPort int
+	EgressOTLPPort  int
+	Port            int
+	Ca              *certtest.Authority
+	Cert            *certtest.Certificate
+}
+
+func NewOTELConfigVars() OTELConfigVars {
+	ingressOTLPPort := 5000 + GinkgoParallelProcess()*100
+	egressOTLPPort := 5000 + GinkgoParallelProcess()*100 + 1
+	port := 5000 + GinkgoParallelProcess()*100 + 2
+
+	ca, err := certtest.BuildCA("otel")
+	Expect(err).NotTo(HaveOccurred())
+	cert, err := ca.BuildSignedCertificate("egress")
```

Comment:
We decided to leave it like it is, they could be separate certs but there's already a lot of TLS boilerplate in the tests and it didn't seem like it would help us catch much that is important. Hopefully the otel-collector doesn't end up with a bug where it's using random certificates in the future.